opentech/apply/models.py

```diff
@@ -1,3 +1,111 @@
-# from django.db import models
+from django.db import models
+from modelcluster.fields import ParentalKey
+from modelcluster.models import ClusterableModel
+from wagtail.wagtailadmin.edit_handlers import (
+    FieldPanel,
+    InlinePanel,
+    StreamFieldPanel,
+)
+from wagtail.wagtailcore.fields import StreamField
+from wagtail.wagtailcore.models import Orderable, Page
+from wagtail.wagtailsearch import index
 
-# Create your models here.
+from opentech.stream_forms.models import AbstractStreamForm
+from opentech.utils.models import SocialFields, ListingFields
+
+from .blocks import CustomFormFieldsBlock
+from .forms import WorkflowFormAdminForm
+from .workflow import SingleStage, DoubleStage
+
+
+WORKFLOW_CLASS = {
+    SingleStage.name: SingleStage,
+    DoubleStage.name: DoubleStage,
+}
+
+
+class ApplyHomePage(Page, SocialFields, ListingFields):  # type: ignore
+    # Only allow creating HomePages at the root level
+    parent_page_types = ['wagtailcore.Page']
+    subpage_types = ['FundPage']
+
+    strapline = models.CharField(blank=True, max_length=255)
+
+    search_fields = Page.search_fields + [
+        index.SearchField('strapline'),
+    ]
+
+    content_panels = Page.content_panels + [
+        FieldPanel('strapline'),
+    ]
+
+    promote_panels = (
+        Page.promote_panels +
+        SocialFields.promote_panels +
+        ListingFields.promote_panels
+    )
+
+
+class FundPage(AbstractStreamForm):
+    parent_page_types = [ApplyHomePage]
+    base_form_class = WorkflowFormAdminForm
+    WORKFLOWS = {
+        'single': SingleStage.name,
+        'double': DoubleStage.name,
+    }
+
+    workflow = models.CharField(choices=WORKFLOWS.items(), max_length=100, default='single')
+
+    def get_defined_fields(self):
+        return self.forms.all()[0].fields
```

Comment:
Adding as a note -- this currently returns the first form fields for a double stage WF fund